index.js

```diff
@@ -204,27 +205,29 @@ Sevnup.prototype._handleRingStateChange = function _handleRingStateChange(arg, d
 Sevnup.prototype._forEachKeyInVNodesWithRetry = function _forEachKeyInVNodesWithRetry(retryErrors, vnodes, onKey, done) {
     var self = this;
 
-    async.eachSeries(vnodes, onVNode, done);
+    async.eachLimit(vnodes, MAX_PARALLEL_TASKS, onVNode, done);
 
     function onVNode(vnode, next) {
         async.waterfall([
             function _loadWithRetries(wNext) {
-                maybeWithRetry("loadkeys", self.loadKeyRetryQueue,
+                _tryWithRetry("loadkeys", self.loadKeyRetryQueue,
                     self.store.loadKeys.bind(self.store, vnode), wNext);
             },
             onKeys.bind(null, vnode),
         ], next);
     }
 
     function onKeys(vnode, keys, next) {
-        async.eachLimit(keys, MAX_PARALLEL_TASKS, function _try(key, eNext) {
-            maybeWithRetry("onkey", self.keyRetryQueue, function _try(cb) {
-                onKey(vnode, key, cb);
-            }, eNext);
+        async.eachLimit(keys, MAX_PARALLEL_TASKS, function _each(key, eachNext) {
+            _tryWithRetry("onkey", self.keyRetryQueue, function _each(cb) {
+                // Instead of calling the key-handler directly, effectively chaining them
+                // we're enqueuing them instead
+                setImmediate(onKey, vnode, key, cb);
+            }, eachNext);
         }, next);
     }
 
-    function maybeWithRetry(retryName, queue, fn, cb) {
+    function _tryWithRetry(retryName, queue, fn, cb) {
```

Comment:
This change seems unnecessary?

Reply:
While its not required strictly, it has been confusing me for a while and just decided to clean it up